### DISCUSSION TITLE
Generate a list using chat models

### DISCUSSION THREAD

**Discussion by abhinavdayal** [guidance-ai/guidance#188] [Q&A]
I have a need to output a list of items from a text (for example list of ingredients from a recipe text) using the ChatGPT model.

How can I use guidance template to specify return the items as a list. Currently it gives me a single text snippet for all ingredients.

```
{{#assistant~}}
{{gen 'ingredients' temperature=0 max_tokens=500}}
{{~/assistant}}
```

**Comment by rickbraddy-pharma** [comment]
{{gen 'ingredients' **n=5** temperature=0 max_tokens=500}}

From [gen notebook](https://github.com/microsoft/guidance/blob/main/notebooks/api_examples/library/gen.ipynb):

n keyword argument
The n argument controls how many generations to perform in a batch. If n > 1 then only the first completion is used for future contex, and the rest are just stored in the variable.

out = guidance("""This is a sentence about {{gen 'text' **n=5** max_tokens=5 temperature=1.0}}""")()
out["text"]
This is a sentence about the 'rightness' of
["the 'rightness' of",
 'a different sort of person,',
 'self destruction in general as the',
 "'our collective intelligence tells us",
 "what we call a 'human"]
